app/org/sagebionetworks/bridge/models/healthdata/HealthDataSubmission.java

```diff
@@ -0,0 +1,120 @@
+package org.sagebionetworks.bridge.models.healthdata;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.joda.time.DateTime;
+
+import org.sagebionetworks.bridge.json.DateTimeDeserializer;
+import org.sagebionetworks.bridge.json.DateTimeSerializer;
+import org.sagebionetworks.bridge.models.BridgeEntity;
+
+/** Health data submitted by the user. */
+@JsonDeserialize(builder = HealthDataSubmission.Builder.class)
+public class HealthDataSubmission implements BridgeEntity {
+    private final String appVersion;
+    private final DateTime createdOn;
+    private final JsonNode data;
+    private final String phoneInfo;
+    private final String schemaId;
+    private final int schemaRevision;
+
+    /** Private constructor. To construct, use builder. */
+    private HealthDataSubmission(String appVersion, DateTime createdOn, JsonNode data, String phoneInfo,
+            String schemaId, int schemaRevision) {
+        this.appVersion = appVersion;
+        this.createdOn = createdOn;
+        this.data = data;
+        this.phoneInfo = phoneInfo;
```

Comment:
nomenclature: is this related to clientInfo? It's possible we'll have non-"phone" (native) clients, like mobile browsers, at some pointing the future.

Reply:
It's actually not. The name comes from the original ResearchKit/AppCore uploads, where they had a field called phoneInfo in the info.json. This is being written to Synapse as phoneInfo.

We could try to merge phoneInfo and ClientInfo (and appVersion) together into a single unified construct, but at this point, it would be more confusing than not, and I think that level of change is outside the scope of this code change.